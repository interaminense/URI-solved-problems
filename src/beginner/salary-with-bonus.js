/**
 * Make a program that reads a seller's name, his/her fixed salary and the sale's total made by
 * himself/herself in the month (in money). Considering that this seller receives 15% over all
 * products sold, write the final salary (total) of this seller at the end of the month , with two
 * decimal places.
 *
 * INPUT
 *
 * The input file contains a text (employee's first name), and two double precision values, that
 * are the seller's salary and the total value sold by him/her.
 *
 * OUTPUT
 *
 * Print the seller's total salary, according to the given example
 */

const tmpl = salary => `TOTAL = ${salary}`;

const salary = (name, fixedSalary, salesTotal) => {
  const totalSalary = salesTotal * 0.15 + fixedSalary;

  return tmpl(totalSalary.toFixed(2));
};

export { tmpl, salary };
