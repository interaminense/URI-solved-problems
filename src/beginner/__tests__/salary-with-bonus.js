import { salary, tmpl } from '../salary-with-bonus';

describe('Salary with bonus', () => {
  let name = '';
  let fixedSalary = 0;
  let salesTotal = 0;
  const result = (a, b, c) => salary(a, b, c);

  it('The fixed salary of JOAO is 500 and sallers total is 1230.30, his bonus is 684.54', () => {
    name = 'JOAO';
    fixedSalary = 500;
    salesTotal = 1230.30;

    expect(result(name, fixedSalary, salesTotal)).toEqual(tmpl(684.54));
    expect(result(name, fixedSalary, salesTotal)).toMatchSnapshot();
  });

  it('The fixed salary of PEDRO is 700.00 and sallers total is 0, his bonus is 700', () => {
    name = 'PEDRO';
    fixedSalary = 700;
    salesTotal = 0;

    expect(result(name, fixedSalary, salesTotal)).toEqual(tmpl('700.00'));
    expect(result(name, fixedSalary, salesTotal)).toMatchSnapshot();
  });

  it('The fixed salary of MANGOJATA is 1700 and sallers total is 1230.50, his bonus is 1884.58', () => {
    name = 'MANGOJATA';
    fixedSalary = 1700;
    salesTotal = 1230.50;

    expect(result(name, fixedSalary, salesTotal)).toEqual(tmpl(1884.58));
    expect(result(name, fixedSalary, salesTotal)).toMatchSnapshot();
  });
});
