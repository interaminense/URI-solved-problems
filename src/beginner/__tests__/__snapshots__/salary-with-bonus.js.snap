// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Salary with bonus The fixed salary of JOAO is 500 and sallers total is 1230.30, his bonus is 684.54 1`] = `"TOTAL = 684.54"`;

exports[`Salary with bonus The fixed salary of MANGOJATA is 1700 and sallers total is 1230.50, his bonus is 1884.58 1`] = `"TOTAL = 1884.58"`;

exports[`Salary with bonus The fixed salary of PEDRO is 700.00 and sallers total is 0, his bonus is 700 1`] = `"TOTAL = 700.00"`;
